model/Licensing/Licensing.md

```diff
@@ -0,0 +1,97 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Licensing
+
+## Summary
+
+The Licensing Profile defines a minimum set of license information required to comply with typical license compliance use cases.
+
+## Description
+
+The Licensing profile only contains additional restrictions.
+Classes and Property restrictions are defined in the SimpleLicensingProfile
+(Classes and Properties associated with string license expressions) and in the ExpandedLicensingProfile (Classes and Properties used for a
+fully parsed syntax tree of license expressions).
+
+There are 2 relationships which are required for any Software Artifact - declaredLicense and concludedLicense.
+
+A declaredLicense is the license identified in text in the software package,
+file or snippet as the license declared by its authors.
+
+This field is not intended to capture license information obtained from an
+external source, such as a package's website. Such information can be
+included, as needed, in a concludedLicense field.
+
+A declaredLicense may be expressed differently in practice for different
+types of artifacts. For example:
+
+* for Packages:
+  * would include license info describing the license of the Package as a
+    whole, when it is found in the Package itself (e.g., LICENSE file,
+    README file, metadata in the repository, etc.)
+  * would not include any license information that is not in the Package
+    itself (e.g., license information from the project’s website or from a
+    third party repository or website)
+* for Files:
+  * would include license info found in the File itself (e.g., license
+    header or notice, comments, SPDX-License-Identifier expression)
+  * would not include license info found in a different file (e.g., LICENSE
+    file in the top directory of a repository)
+* for Snippets:
+  * would include license info found in the Snippet itself (e.g., license
+    notice, comments, SPDX-License-Identifier expression)
+  * would not include license info found elsewhere in the File or in a
+    different File (e.g., comment at top of File if it is not within the
+    Snippet, LICENSE file in the top directory of a repository)
+
+If a declaredLicense has a NONE value (NoneLicense), this indicates that the
+corresponding Package, File or Snippet contains no license information
+whatsoever.
+
+If a declaredLicense has a NOASSERTION value (NoAssertionLicense), this
+indicates that one of the following applies:
+* the SPDX data creator has attempted to but cannot reach a reasonable
+  objective determination;
+* the SPDX data creator has made no attempt to determine this field; or
+* the SPDX data creator has intentionally provided no information (no meaning
+  should be implied by doing so).
+  
+If a declaredLicense is not present or a null value, no conclusion can be drawn.
```

Comment:
@maxhbr - I would leave it as null - I don't think we're changing the semantics?

Reply:
for 2.3 NOASSERTION is equal to `null`, since there was no `null` in the model. We are introducing it with 3.0. So we should treat an explicit NOASSERTION equal to the shorthand `null`.

Reply:
> for 2.3 NOASSERTION is equal to null, since there was no null in the model. 

In 2.3 we made the license properties optional, so I believe they can be null.  That is how I'm interpreting a missing value in the JSON file.

If we have a JSON file where a license field is not present in 2.3, we should likewise have it not present in 3.0.  I don't see a need to convert it to a NOASSERTION.

Is the issue with the text `or a null value`?  If so, we can remove it.

Reply:
Sorry for going in circles here. In 2.3 the properties were optional (cardinality 0 to 1) but the following text was added to the description:

> If the Declared License field is not present for a package, it implies an equivalent meaning to NOASSERTION. 

this effectively collapses the the "not present" and the "NOASSERTION" case again, for 2.3. In my recollection this was purely added to allow for a shorter serialization and the Intend was not to make the model more expressive.

Reply:
Got it.

I don't know if we intentionally changed the semantics of declared license.  I still think if there is no value in 2.3, we should create a `NoAssertionLicense` relationship in 3.0.

@swinslow - any thoughts?

Reply:
agree, mapping  "not present" and the "NOASSERTION" both to the `NoAssertionLicense` Relationship is consistent for me.

Reply:
Note that in 3.0 the author of SPDX data can still choose not to have a declared license, so not mapping it to a `NoAssertionLicense` would be OK too.  We just need to decide what to document in the migration guide.

I have a call with @swinslow this afternoon to sync up - I'll check with Steve and update the issue.

@maxhbr - Since this is specific to 2.3 migration and would only be documented in the migration guide IMO, can you approve this PR?

Reply:
This raises a question: I'm not sure what "not present" means, now that Declared License is a Relationship rather than a property. "Not present" in this particular SPDX document, BOM, bundle, namespace, ...? (Apologies as I'm not certain which term to use here!)

I believe the purpose of decoupling license values from the underlying software artifacts was in part to make it so that anyone could more easily create licensing values (via Relationships) for software artifacts that someone else had defined as SPDX Elements previously.

Given that, I'm not sure that a "null value" really has relevance anymore, unless I'm misunderstanding. Perhaps this line should just be deleted.

Reply:
>This raises a question: I'm not sure what "not present" means, now that Declared License is a Relationship rather than a property. "Not present" in this particular SPDX document, BOM, bundle, namespace, ...? (

Couple of thoughts:
- Not present would just indicate that there is no `declaredLicense` relationship type found from a `SoftwareArtifact` to a license - no matter what was serialized - if you don't find it, it isn't present.
- There is no `null` value in the model itself - either a relationship is present or it isn't.  There is a null value if we treat it as a property.

I'll remove the null value reference.